packages/layers/src/multiscale-image-layer/multiscale-image-layer.js

```diff
@@ -7,6 +7,50 @@ import ImageLayer from '../image-layer';
 import { getImageSize, isInterleaved, SIGNAL_ABORTED } from '@vivjs/loaders';
 import { ColorPaletteExtension } from '@vivjs/extensions';
 
+/**
+ * Here we create an object for checking clipping the black border of incoming tiles
+ * at low resolutions i.e for zarr tiles which are padded by zarr.
+ * We need to check a few things before trimming:
+ *  1. The height/width of the full image at the current resolution
+ *  produces an image smaller than the current tileSize
+ *  2. The incoming image is padded to the tile size i.e one of its dimensions matches the tile size
+ * Once these have been confirmed, we trim the tile by going over it in row major order,
+ * keeping only the data that is not out of the clipped bounds.
+ * @param {{
+ *   loader: PixelSource[],
+ *   resolution: number,
+ *   tileSize: number,
+ * }}
+ * @return {{ clip: function, height: number, width: number }}
+ */
+const createTileClipper = ({ loader, resolution, tileSize }) => {
+  const planarSize = Object.values(getImageSize(loader[0]));
+  const [clippedHeight, clippedWidth] = planarSize.map(size =>
+    Math.floor(size / 2 ** resolution)
+  );
```

Comment:
This relies on the ordering of key names in `getIamgeSize`, if those were to change this could break.

```suggestion
  const planarSize = getImageSize(loader[0]);
  const [clippedHeight, clippedWidth] = [planarSize.height, planarSize.width].map(size =>
    Math.floor(size / 2 ** resolution)
  );
```